### ISSUE TITLE
[downloads-pro] downloads: enabled: false is always set

### ISSUE THREAD

**Issue by aleclerc7** [getgrav/grav-premium-issues#240]
Hello,

Even if downloads are deactivated in a page (default value, false), it is always written in the .md file.

I would like to suggest the if the value is false, to not write in in the .md files that are being modified for other purposes.

Just modifying a title will cause `downloads: enabled: false` to be written.

Kind regards.

**Comment by mahagr** [comment]
This is really how the saving works everywhere in the admin. Unless the value is an empty string, it will always be saved.

That said, we could try to make the value empty, where it would only save the `true` value...?

**Comment by aleclerc7** [comment]
Or, just like the other fields which have a check box to really say, “Yes, I did set that value for this page.”

Right now I can tell you all the pages I saved since I bought Downloads Pro. It adds its mark everywhere. 😊

Kind regard.

**Comment by rhukster** [comment]
We could look into using 'toggleable' fields.  That way they don't send any data unless 'toggled'.

**Comment by rhukster** [comment]
Just tested this and while it works for pages by default, (ie, just saving on a page with no downloads).  As soon as you toggle yes, you get values stored, and even disabling won't remove those values under `downloads:`.  I think its better than nothing though. At least if you never enable it, it never pollutes the frontmatter.

**Comment by aleclerc7** [comment]
I thought I sent a message concerning that. Indeed, I tried with toggleable, and the problem is that all the other options are checked if one does so. This is not very good since many default values are fine just like that. Maybe I made another suggestion, I do not remember, but the problem is that the toggled value should not be inherited by children of the field.

Anyway. To implement this suggestion properly, some other code has to be fixed or changed.

Kind regards.
